src/main/java/liquibase/ext/clickhouse/sqlgenerator/CreateDatabaseChangeLogLockTableClickHouse.java

```diff
@@ -40,20 +43,46 @@ public boolean supports(CreateDatabaseChangeLogLockTableStatement statement, Dat
     return database instanceof ClickHouseDatabase;
   }
 
+  private String generateSqlTemplate(boolean replicated) {
```

Comment:
IMO I think it would be simpler to separate generation into sections (generateEngineClause, etc) because the conditions would be applied in one place (if isReplicated)